src/index.ts

```diff
@@ -991,6 +1005,14 @@ function readConfig (
     useCaseSensitiveFileNames: ts.sys.useCaseSensitiveFileNames
   }, basePath, undefined, configFileName))
 
+  if (tsconfigOptions.require) {
```

Comment:
Thoughts on doing the require for these options within `index.ts` so configuration flows in a single direction? It might make it easier instead of needing to resolve and pass it back to the CLI which requires. Also, this would ensure we cover the programmatic usage of `register()` and via `ts-node/register`.

Reply:
Yeah I like that.

The CLI uses `Module._preloadModules`; do you know if that function is doing anything special that will break if we switch to a normal `require()` call?  There's nothing wrong with `_preloadModules` but I don't really understand what it does other than `require()` stuff.

Reply:
Yeah, I recall I tried to change it one time. Feel free to keep the CLI logic as it is and build the internal one separately for now so there's no regression. Here's the implementation in node.js: https://github.com/nodejs/node/blob/ec2ffd6b9d255e19818b6949d2f7dc7ac70faee9/lib/internal/modules/cjs/loader.js#L1400-L1417. 

Here's where I had changed it at one point, but I can't seem to find the issue. I should have linked things better in the past 🤕 https://github.com/TypeStrong/ts-node/commit/2e99c5065f07293bb40649029990498515ba9616#diff-ee45801c137c3b9f9d8ec7ece340a514R197.

Reply:
The main thing it's trying to ensure is that the `require()` is occurring from the place where you run the script, not where `ts-node` lives.

Reply:
Ok, I don't think there's anything wrong with calling `_preloadModules` from within `register()`, so I'll keep doing that.

Should it happen in `register()` or in `create()`.  I feel like `register()` makes more sense because that's where we "install" the environment.  But source map support is installed in `create()`.